# Copyright (c) 2006, 2007, Oracle and/or its affiliates. All rights reserved.
#
# Permission is hereby granted, free of charge, to any person obtaining a
# copy of this software and associated documentation files (the "Software"),
# to deal in the Software without restriction, including without limitation
# the rights to use, copy, modify, merge, publish, distribute, sublicense,
# and/or sell copies of the Software, and to permit persons to whom the
# Software is furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice (including the next
# paragraph) shall be included in all copies or substantial portions of the
# Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL
# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
# DEALINGS IN THE SOFTWARE.
6370961  - needs sane vert, horiz ranges

diff -urp -x '*~' -x '*.orig' src/vmware.c src/vmware.c
--- src/vmware.c	2009-05-12 16:47:50.000000000 -0700
+++ src/vmware.c	2009-05-12 21:01:29.909838000 -0700
@@ -927,6 +927,20 @@ VMWAREPreInit(ScrnInfoPtr pScrn, int fla
      */
     vmwareGetSupportedModelines(&pScrn->monitor->Modes);
 
+    /* Set reasonable default Hsync range, if needed. */
+    if (pScrn->monitor->nHsync == 0) {
+        pScrn->monitor->hsync[0].lo = 31.5f;
+        pScrn->monitor->hsync[0].hi = 64.0f;
+        pScrn->monitor->nHsync = 1;
+    }
+
+    /* Set reasonable default Vsync range, if needed. */
+    if (pScrn->monitor->nVrefresh == 0) {
+        pScrn->monitor->vrefresh[0].lo = 50.0f;
+        pScrn->monitor->vrefresh[0].hi = 70.0f;
+        pScrn->monitor->nVrefresh = 1;
+    }
+
     i = xf86ValidateModes(pScrn, pScrn->monitor->Modes, pScrn->display->modes,
                           clockRanges, NULL, 256, pVMWARE->maxWidth, 32 * 32,
                           128, pVMWARE->maxHeight,
